### PR TITLE
Add checkstyle checks

### LINKED ISSUE #445: Add checkstyle checks
We have to setup checkstyle with the same config of Blobit project.
https://github.com/diennea/blobit

### PR DESCRIPTION
**Pull request by TisonKun**
Checkstyle file is copied from [blobit](https://github.com/diennea/blobit), with following changes

1. Didn't forbid `TODO` comments. `FIXME|XXX` are still forbidden, but given that there are several `TODO`s and I see no reason to just turn in another word, I loose the rule in someway.
2. `(Class|Method|Interface)TypeParameterName` are losen to allow `[A-Z]` since I notice there are several usages of `P`/`O`/`X`/`Y` and we don't have too many reason to change it.
3. Comment out `LineLength` rule. We should have another pass to do it. Unfortunately it's too expensive to me to include in this pull request.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

close #445

### CODE REVIEW

File: herddb-core/src/test/java/herddb/index/blink/BlinkBench.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
 import java.io.PrintStream;
 import java.io.PrintWriter;
 import java.io.StringWriter;
-import java.nio.charset.StandardCharsets;
+import java.io.UnsupportedEncodingException;
```
Comment (TisonKun) [inline herddb-core/src/test/java/herddb/index/blink/BlinkBench.java]: Revert changes introduced by code inspector. The method is added in Java 10 and we have to support 1.8 though, although we're in test scope.

**Comment by TisonKun** [comment]
travis fails on `ConcurrentCheckpointBLinkTest.concurrentReadWithManyEmptyNodes`

cannot reproduce locally.

**Comment by eolivelli** [comment]
Great work!
I have restarted Travis, usually we don't have such flaky tests, and your change did not change the behavior.

We will review carefully and merge as soon as possible

Thank you very much

Welcome to this project!

**Comment by eolivelli** [comment]
Please state in a comment in this PR that your contribution follows the rule of the Apache 2 license, as this is your first contrib and it is an huge commit

**Comment by TisonKun** [comment]
Update pull request description and commit message.

**Pull request merged**